tests/nested/image-build/task.yaml

```diff
@@ -2,58 +2,73 @@ summary: create ubuntu-core image and execute the suite in a nested qemu instanc
 
 systems: [ubuntu-16.04-64]
 
-environment:
-    SSH_PORT: 8022
-    USER: user1
-    PASSWORD: ubuntu
-    SPREAD_EXTERNAL_ADDRESS: "localhost:8022"
-    WORK_DIR: /tmp/work-dir
-    TAP_DEVICE: tap0
-    VM_UNIT_COMMAND_SUFFIX: "-m 1024 -nographic -net nic,model=virtio -net user,hostfwd=tcp::${SSH_PORT}-:22 -drive file=${WORK_DIR}/ubuntu-core.img,if=virtio,cache=none -drive file=${PWD}/assertions.disk,if=virtio,cache=none"
-
 prepare: |
-    . $TESTSLIB/nested.sh
-
     # determine arch related vars
-    set_vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
```

Comment:
Indeed, that was the original idea and what would give us more value from all this approach, I've spent some time with the pi2 emulator with no luck so far, will keep trying anyway :)